web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -109,7 +114,7 @@
     @RequestMapping(value = "/clinical-data-bin-counts/fetch", method = RequestMethod.POST,
         consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Fetch clinical data bin counts by study view filter")
-    public ResponseEntity<List<DataBin>> fetchClinicalDataBinCounts(
+    public ResponseEntity<List<ClinicalDataBin>> fetchClinicalDataBinCounts(
```

Comment:
This method is huge.  In general, I think controller methods should just apply permissions and validate arguments.   Is there a way to separate the validation of a filter with the apply of a filter?   Also, I feel the calculation of data bins should happen at a lower layer.  Is there a reason why we never introduced a StudyViewService?